app/crud.py

```diff
@@ -69,9 +69,23 @@ def create_step(db: Session,
         if key == "script_file":
             db_step.script_file = value
         elif key == "input":
-            db_step.input = dir_head + value + dir_tail
+            if isinstance(value, str):
+                # This is a simple input field.
+                db_step.input = dir_head + value + dir_tail
+            elif value.get("no_batch_in_path"):
+                db_step.input = dir_head + value[key]
+            else:
+                raise(ValueError(f"In the config.yaml the input was "
+                                 f"in an unknown structure"))
         elif key == "output":
-            db_step.output = dir_head + value + dir_tail
+            if isinstance(value, str):
+                # This is a simple output field.
+                db_step.output = dir_head + value + dir_tail
+            elif value.get("no_batch_in_path"):
+                db_step.output = dir_head + value[key]
+            else:
+                raise (ValueError(f"In the config.yaml the output was "
+                                  f"in an unknown structure"))

```

Comment:
The input and output setup is basically the same, might make sense to extract the code into a separate function.

Reply:
Ok, will do.